From 0fc80a02c8d3596c1f78dd1fb75b1f4e34788d38 Mon Sep 17 00:00:00 2001
From: Wright Feng <wright.feng@cypress.com>
Date: Wed, 27 Feb 2019 01:29:27 -0600
Subject: [PATCH 100/118] brcmfmac: allocate msgbuf pktid from 1 to size of
 pktids array

Some PCIE firmwares drop txstatus with pktid=0, that makes packet stays in
host side and not release. If the packet is 8021x packet, the
pend_8021x_cnt will be always larger than 0 and see "Timed out waiting
for no pending 802.1x packets" error when sending key to dongle.

Signed-off-by: Wright Feng <wright.feng@cypress.com>

---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/msgbuf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/msgbuf.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/msgbuf.c
index d2c834c..1a7b203 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/msgbuf.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/msgbuf.c
@@ -330,7 +330,7 @@ struct brcmf_msgbuf_pktids {
 	do {
 		(*idx)++;
 		if (*idx == pktids->array_size)
-			*idx = 0;
+			*idx = 1;
 		if (array[*idx].allocated.counter == 0)
 			if (atomic_cmpxchg(&array[*idx].allocated, 0, 1) == 0)
 				break;
-- 
1.9.1

